core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailDataset.java

```diff
@@ -94,4 +118,12 @@ default CloseableIteration<? extends Triple> getTriples(Resource subj, IRI pred,
 		throw new SailException("RDF-star triple retrieval not supported by this store");
 	}
 
+	default Set<StatementOrder> getAvailableOrderings(Resource subj, IRI pred, Value obj, Resource... contexts) {
+		return Set.of();
+	}
```

Comment:
A way for the SailDataset to inform upstream code about which orderings it can provide for different statement patterns.